includes/checkout.php

```diff
@@ -291,9 +285,90 @@ function pmprogl_pmpro_after_checkout($user_id, $morder)
+	global $current_user;
+
+	// Get the gift code from the database.
+	$code = $wpdb->get_var("SELECT code FROM $wpdb->pmpro_discount_codes WHERE id = '" . intval( $code_id ) . "' LIMIT 1");
+
+	// Set some empty variables if not set previously.
+	if ( ! isset( $recipient_email ) ) {
+		$recipient_email = '';
+	}
+	if ( ! isset( $gift_message ) ) {
+		$gift_message = '';
+	}
+
+	// Build the email template variable replacement data.
+	$data = array(
+		// Gift Membership data.
+		'pmprogl_gift_recipient_email' => $recipient_email,
+		'pmprogl_giver_display_name' => $current_user->display_name,
+		'pmprogl_gift_message' => wp_unslash( $gift_message ),
+		'pmprogl_gift_code' => $code,
+		'pmprogl_gift_code_url' => pmpro_url( 'checkout', '?level=' . intval( $gift['level_id'] ) . "&discount_code=" . $code ),
+
+		// Order data.
+		'invoice_id' => $morder->code,
+		'invoice_total' => pmpro_formatPrice($morder->total),
+		'invoice_date' => date_i18n(get_option('date_format'), $morder->getTimestamp()),
+		'billing_name' => $morder->billing->name,
+		'billing_street' => $morder->billing->street,
+		'billing_city' => $morder->billing->city,
+		'billing_state' => $morder->billing->state,
+		'billing_zip' => $morder->billing->zip,
+		'billing_country' => $morder->billing->country,
+		'billing_phone' => $morder->billing->phone,
+		'cardtype' => $morder->cardtype,
+		'accountnumber' => hideCardNumber($morder->accountnumber),
+		'expirationmonth' => $morder->expirationmonth,
+		'expirationyear' => $morder->expirationyear,
+		'billing_address' => pmpro_formatAddress($morder->billing->name,
+												 $morder->billing->street,
+												 "", //address 2
+												 $morder->billing->city,
+												 $morder->billing->state,
+												 $morder->billing->zip,
+												 $morder->billing->country,
+												 $morder->billing->phone),
+		'invoice_url' => pmpro_login_url( pmpro_url( 'invoice', '?invoice=' . $morder->code ) ),
+	);
+
+	// Send email to the gift purchaser.
+	$email_purchaser = new PMProEmail();
+	$email_purchaser->template = 'pmprogl_gift_purchased';
+	$email_purchaser->email = $current_user->email;
+	$email_purchaser->data = $data;
+	$email_purchaser->sendEmail();
+
+	// Send email to the site admin.
+	$email_admin = new PMProEmail();
+	$email_admin->template = 'pmprogl_gift_purchased_admin';
+	$email_admin->email = get_bloginfo( 'admin_email' );
+	$email_admin->data = $data;
+	$email_admin->sendEmail();
+
+	// Send email to the gift recipient (optional).
+	if ( ! empty( $recipient_email ) ) {
+		// Replace any default email data that is user-specific.
+		$new_data = array(
+			'name' => $recipient_email,
+			'display_name' => $recipient_email,
+			'user_email' => $recipient_email,
+		);
+		$email_recipient = new PMProEmail();
+		$email_recipient->template = 'pmprogl_gift_recipient';
+		$email_recipient->email = $recipient_email;
+		$email_recipient->data = array_merge( $data, $new_data );
+		$email_recipient->sendEmail();
+	}
+
```

Comment:
Probably not going to matter, but may make sense to put this all in the `if($wpdb->query($sqlQuery) !== false)` statement above so that we know that a discount code was successfully created before sending an email and such. We also wouldn't have to get $code again from the DB.

Reply:
I think it would be beneficial to move these into some functions that do the calls, that would make moving it into the `if($wpdb->query($sqlQuery) !== false)` area when it inserts the code -- much more easier to read.

Reply:
I agree, moving into functions make sense. I'm going to merge this PR and then work on this change

Reply:
Cleaned up this code here: https://github.com/strangerstudios/pmpro-gift-levels/commit/169933ddec9a2e21a45bda533e50862436467077